src/vsr/superblock.zig

```diff
@@ -819,10 +812,10 @@ pub fn SuperBlockType(comptime Storage: type) type {
             update: UpdateCheckpoint,
         ) void {
             assert(superblock.opened);
-            assert(update.commit_min <= update.commit_max);
-            assert(update.commit_min > superblock.staging.vsr_state.checkpoint.commit_min);
-            assert(update.commit_min_checksum !=
-                superblock.staging.vsr_state.checkpoint.commit_min_checksum);
```

Comment:
I would add

```
assert(update.header.valid_checksum());
assert(update.header.command == .prepare);
```

Reply:
Good call, added this as well as `.command` check to the `on_sync_checkpoint`